### DISCUSSION TITLE
How to add new Board with PlatformIO Support?

### DISCUSSION THREAD

**Discussion by derlucas** [zephyrproject-rtos/zephyr#31393] [Q&A] [answered]
Please see my question here in the PIO discussion groups:
https://community.platformio.org/t/try-to-add-new-board-with-zephyr-support-dtsi-file-missing/18581/4

I'm looking for a reason why my local board config in PIO is not working. Why is the file not found with the local config, but working with the "steval_fcu001v1" board.

Any help or hints where i should look are apprechiated.

**Comment by martinjaeger** [comment, marked as answer]
The problem is that you are using the new pinmux features for STM32 introduced in Zephyr 2.5, but PlatformIO is using current stable release 2.4.

A possible workaround requires a custom fork of the `stm32_hal` repo which contains a `package.json` as required by PlatformIO. Afterwards you can point PlatformIO towards that repo. I have done that already for the [Libre Solar Charge Controller Firmware develop branch](https://github.com/LibreSolar/charge-controller-firmware/tree/develop). So it should work if you just add the following lines to `platformio.ini`:

```ini
platform_packages =
    framework-zephyr-hal-stm32@https://github.com/LibreSolar/hal_stm32#v2.5-platformio
```

Otherwise just take board files from Zephyr `v2.4-branch` as the basis for your local board config. The Zephyr `master` branch is always work in progress.
